### PR TITLE
feature(stories migrate): Migrate Calendar stories to new architecture

### PR DESCRIPTION
**Pull request by vekkele**
https://www.notion.so/santiment/Stories-migration-Calendar-a7a563e09f034006a8ab03bcc0bd0f55?pvs=4

### CODE REVIEW

File: src/ui/Calendar/PresetCalendar.svelte [hunk 1]
```diff
@@ -35,10 +39,10 @@
     },
   ]
 
-  export let date = [new Date(), new Date()]
+  export let date: DateRange = [new Date(), new Date()]
```
Comment (DmitriVanGuard) [inline src/ui/Calendar/PresetCalendar.svelte:42]: Initial `date` property is required to have tuple of `2` dates, due to `range` calendar mode. If tuple with a single date is passed, calendar will await selection of the second date, which might be considered not good UX.
Solution: strict type `[Date, Date]`

**Review by DmitriVanGuard — CHANGES_REQUESTED** [review]

File: src/ui/Calendar/PresetCalendar.svelte (hunk 1 above)
Comment (vekkele) [reply src/ui/Calendar/PresetCalendar.svelte:42]: Fixed

File: src/ui/Calendar/InputCalendar.svelte [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+  import type { DateRange } from './PresetCalendar.svelte'
+
   import Svg from '@/ui/Svg/svelte'
   import PresetCalendar from './PresetCalendar.svelte'
   import { getDateFormats, MONTH_NAMES, setDayStart, setDayEnd } from '@/utils/dates'
 
-  export let date = [new Date(), new Date()] as [Date, Date]
-  export let onDateSelect: (date: Date[]) => void
+  export let date: DateRange = [new Date(), new Date()]
+  export let onDateSelect: (date: DateRange) => void
```
Comment (DmitriVanGuard) [inline src/ui/Calendar/InputCalendar.svelte:11]:
```suggestion
  export let onDateSelect: (date: Date[]) => void
```

File: src/ui/Calendar/PresetCalendar.svelte [hunk 3]
```diff
@@ -35,10 +39,10 @@
...
   export let label = date[0].toLocaleDateString()
   export let maxDate: Date = new Date()
-  export let onDateSelect: (date: Date[]) => void
+  export let onDateSelect: (date: DateRange) => void
```
Comment (DmitriVanGuard) [inline src/ui/Calendar/PresetCalendar.svelte:45]:
```suggestion
  export let onDateSelect: (date: Date[]) => void
```

**Review by DmitriVanGuard — APPROVED** [review]

**Pull request merged**